### PR TITLE
[add] Search behavior

### PR DESCRIPTION
**Pull request by nhunzaker**
Adds search behavior with a slight delay between key presses and abortion of previous requests.

### CODE REVIEW

File: src/mixins/sync.js [hunk 1]
```diff
@@ -11,24 +11,39 @@ let Types = React.PropTypes
 export default {
 
   propTypes: {
-    onFetch : Types.func
+    buildQuery : Types.func,
```
Comment (solomonhawk) [inline src/mixins/sync.js]: Should this be `urlBuilder`?

**Comment by solomonhawk** [comment]
Tests? ;)

**Comment by tommymarshall** [comment]
Good stuff with `request.abort()` :+1:

File: src/mixins/sync.js (hunk 1 above)
Comment (nhunzaker) [inline src/mixins/sync.js]: probably

**Comment by nhunzaker** [comment]
@solomonhawk Heh, that's valid Sol.

**Comment by nhunzaker** [comment]
There you go, Sol  cf4212b. I'll do a more comprehensive follow up

**Pull request merged**